cli/src/main.rs

```diff
@@ -258,6 +269,20 @@ fn main() -> Result<()> {
     std::fs::create_dir_all(directory).context(format!("{directory:?} could not be created"))?;
 
     let log_path = directory.join(logname);
+
+    Ok(log_path)
+}
+
+fn main() -> Result<()> {
+    // Parse the commandline options with structopt.
+    let matches = Cli::parse();
+
+    let log_path = if let Some(location) = matches.log_file {
```

Comment:
How about we make this check oart of the `default_logfile_location()` function and just call it `logfile_location`?

Reply:
I prefer it like this, otherwise we just pass a parameter which then gets immediately returned.